prow/plugins/trigger/generic-comment.go

```diff
@@ -158,11 +160,21 @@ func handleGenericComment(c Client, trigger plugins.Trigger, gc github.GenericCo
 					runID := run.ID
 					go func() {
 						if err := c.GitHubClient.TriggerFailedGitHubWorkflow(org, repo, runID); err != nil {
-							log.Errorf("attempt to trigger github run failed: %v", err)
+							log.Errorf("attempt to trigger github run failed for %v: %v", run.Name, err)
+							if strings.Contains(err.Error(), "Unable to retry this workflow run because it was created over a month ago") {
+								expiredWorkflows = append(expiredWorkflows, run.Name)
```

Comment:
@petr-muller / @stevekuznetsov do either of you remember why we are making calls to retrigger workflows in separate goroutines here?
Original PR: https://github.com/kubernetes/test-infra/pull/24726

Reply:
I don't, I wasn't that involved in that PR. Looking at the commits it seems like an opportunistic way to lower response latency because we only log errors but do not use them in the response? Goroutines were used right in the first [commit](https://github.com/kubernetes/test-infra/pull/24726/commits/1e4703ddd7ec3289eebd13800b3001d319c7adb9#diff-7271c7d0bdb752d0280ebeee09398796ec716e31f7174c98a4fbd6f496eba047R153-R157) so they were likley not added there as a result of a review (there are separate commits that addressed reviews).

Reply:
But yeah, that also means the `expiredWorklows` cannot be used this way, as we can easily hit L172 before the goroutines finish their client calls.

/hold

Reply:
fixed that, i missed that thanks for spotting

Reply:
do you think if I we go with this would we be able to ensure safety and also lower down the latency?

```
var expiredWorkflows []string
var wg sync.WaitGroup
var mu sync.Mutex

for _, run := range failedRuns {
    log := c.Logger.WithFields(logrus.Fields{
        "runID":   run.ID,
        "runName": run.Name,
        "org":     org,
        "repo":    repo,
    })
    wg.Add(1)
    go func(runName string, runID int64, log *logrus.Entry) {
        defer wg.Done()

        if err := c.GitHubClient.TriggerFailedGitHubWorkflow(org, repo, runID); err != nil {
            log.Errorf("attempt to trigger github run failed for %v: %v", runName, err)
            if strings.Contains(err.Error(), "Unable to retry this workflow run because it was created over a month ago") {
                mu.Lock()
                expiredWorkflows = append(expiredWorkflows, runName)
                mu.Unlock()
            }
        } else {
            log.Infof("successfully triggered action run")
        }
    }(run.Name, run.Name, log)
}

wg.Wait()

if len(expiredWorkflows) > 0 {
    comment := fmt.Sprintf(`The following workflows could not be re-triggered because they are now expired (created over a month ago):
- %s
`, strings.Join(expiredWorkflows, "\n- "))
    c.Logger.Infof("Commenting \"%s\".", comment)
    c.GitHubClient.CreateComment(org, repo, number, plugins.FormatResponseRaw(gc.Body, gc.HTMLURL, gc.User.Login, comment))
}

```

Reply:
@AvineshTripathi yes - that should be okay 👍🏼

Reply:
Please also take a look at the test failures.